### PR TITLE
Add EIP1559 Fee management methods and configs

### PR DESCRIPTION
**Pull request by smk762**
PR includes all changes related to the EIP1559 update, including new methods and config params.
Related: https://github.com/KomodoPlatform/komodo-defi-framework/pull/2051

Closes:
- https://github.com/KomodoPlatform/komodo-docs-mdx/issues/286
- https://github.com/KomodoPlatform/komodo-docs-mdx/issues/290
- https://github.com/KomodoPlatform/komodo-docs-mdx/issues/292

### CODE REVIEW

**Comment by smk762** [comment]
Note:
> Plz be aware that gas fee estimator rpcs
> 
> ```
> start_eth_fee_estimator
> stop_eth_fee_estimator
> ```
> 
> will be removed in this PR [2172](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172) as it will be managed by the new streaming API rpc "fee_estimator::enable"

**Comment by smk762** [comment]
@dimxy Should we PR to coins repo for setting ` "gas_fee_estimator": "simple"` by default?

**Comment by dimxy** [comment]
> @dimxy Should we PR to coins repo for setting ` "gas_fee_estimator": "simple"` by default?

Yes currently it is configured in the coins file and should be set explicitly.
But this code is deprecated and will be replaced by a more general [event streaming](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172) feature. I prepared an update for gas fee estimator into the event streaming, in that update fee estimator is configured via the coin activation request (instead of coins file) and yes, it will default to the 'simple' option. I will create this fix as a PR when the event streaming PR is merged.

**Review by dimxy — APPROVED** [review]
LGTM
(with the remark that start_eth_fee_estimator stop_eth_fee_estimator rpcs will be removed when [event streaming PR](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172) is merged)

File: src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx [hunk 1]
```diff
@@ -62,3 +67,47 @@ The `InputTxns` object includes the following items:
...
   ```
 </CollapsibleSection>
+
+### PayForGas
+
+The `PayForGas` object includes the following items:
+
+| Parameter                    | Type    | Description                                                                                                                                                                  |
+| ---------------------------- | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| tx\_type                     | string  | ERC20-like coins and tokens only. Options are `Legacy` or `Eip1559`. The type of transaction values being configured.                                                        |
```
Comment (dimxy) [inline src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx:77]: Not only ERC20
maybe 'ETH-like' (or EVM-like if 'EVM' is well known)

File: src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx [hunk 2]
```diff
@@ -62,3 +67,47 @@ The `InputTxns` object includes the following items:
...
+| gas\_price                   | decimal | Values are in Gwei. The base gas price for the transaction.                                                                                                                  |
```
Comment (dimxy) [inline src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx:78]: gas_price is for tx_type = Legacy, this is max price per gas unit the user is willing to pay for the transaction

File: src/pages/komodo-defi-framework/api/v20/fee_management/get_swap_transaction_fee_policy/index.mdx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+export const description =
+  "The get_swap_transaction_fee_policy method allows you to get the current gas fees policy used when estimating swap transaction fees.";
+
+# get\_swap\_transaction\_fee\_policy
+
+The `get_swap_transaction_fee_policy` method allows you to get the current gas fees policy used when estimating swap transaction fees.
+
+| parameter | Type   | Description                                                            |
+| --------- | ------ | ---------------------------------------------------------------------- |
+| coin      | string | Ticker of the coin/asset for which we want to view the fee policy for. |
```
Comment (dimxy) [inline src/pages/komodo-defi-framework/api/v20/fee_management/get_swap_transaction_fee_policy/index.mdx:11]: currently supported for ETH/EVM coins/tokens

File: src/pages/komodo-defi-framework/api/v20/fee_management/set_swap_transaction_fee_policy/index.mdx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+export const title = "Komodo DeFi Framework Method: Set Swap Transaction Fee Policy";
+export const description =
+  "The set_swap_transaction_fee_policy method allows you to set the gas fee policy used when estimating swap transaction fees.";
+
+# set\_swap\_transaction\_fee\_policy
+
+The `set_swap_transaction_fee_policy` method allows you to set the gas fee policy used when estimating swap transaction fees.
```
Comment (dimxy) [inline src/pages/komodo-defi-framework/api/v20/fee_management/set_swap_transaction_fee_policy/index.mdx:7]: currently supported for ETH/EVM coins/tokens

File: src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx (hunk 1 above)
Comment (smk762) [reply src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx:77]: fixed, thanks

File: src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx [hunk 5]
```diff
@@ -50,7 +56,6 @@ The `InputTxns` object includes the following items:
 | index            | integer | The \[output index([https://bitcoin.stackexchange.com/questions/100765/what-does-the-index-of-an-utxo-stand-for](https://bitcoin.stackexchange.com/questions/100765/what-does-the-index-of-an-utxo-stand-for))] of this unspent transaction output. |
 | script\_pub\_key | string  | The [scriptpubkey](https://learnmeabitcoin.com/technical/scriptPubKey) of this unspent transaction output.                                                                                                                                          |
 | amount           | float   | The value of this unspent transaction output.                                                                                                                                                                                                       |
```
[the remark this answers is not in the document]
Comment (smk762) [reply src/pages/komodo-defi-framework/api/common_structures/wallet/index.mdx:58]: fixed, thanks

**Pull request merged**